Build failure: plugdata

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

add plugdata and rebuild

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/317895965

### Relevant log output

```console
error: builder for '/nix/store/gknhhzwaxln6i36mfv921zpr6b164ihr-plugdata-0.9.1.drv' failed with exit code 2;
       last 25 log lines:
       > /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/lib/gcc/x86_64-unknown-linux-gnu/15.2.0/include/xmmintrin.h:982:23: warning: array subscript '__m128_u[0]' is partly outside array bounds of 'float [2]' [-Warray-bounds=]
       >   982 |   return *(__m128_u *)__P;
       >       |                       ^~~
       > /build/source/Libraries/pd-else/Code_source/Compiled/audio/sfz~/sfizz/library/src/sfizz/effects/Rectify.cpp: In member function 'virtual void sfz::fx::Rectify::process(const float* const*, float* const*, unsigned int)':
       > /build/source/Libraries/pd-else/Code_source/Compiled/audio/sfz~/sfizz/library/src/sfizz/effects/Rectify.cpp:73:23: note: object 'out2x' of size 8
       >    73 |                 float out2x[2];
       >       |                       ^~~~~
       > [ 95%] Building CXX object library/src/CMakeFiles/sfizz_internal.dir/sfizz/simd/HelpersSSE.cpp.o
       > [ 97%] Building CXX object library/src/CMakeFiles/sfizz_internal.dir/sfizz/simd/HelpersAVX.cpp.o
       > [ 97%] Linking CXX static library ../lib/libsfizz_internal.a
       > [ 97%] Built target sfizz_internal
       > [ 98%] Building CXX object library/src/CMakeFiles/sfizz_import.dir/sfizz/import/sfizz_import.cpp.o
       > [ 98%] Building CXX object library/src/CMakeFiles/sfizz_static.dir/sfizz/sfizz.cpp.o
       > [ 98%] Building CXX object library/src/CMakeFiles/sfizz_import.dir/sfizz/import/ForeignInstrument.cpp.o
       > [ 98%] Building CXX object library/src/CMakeFiles/sfizz_static.dir/sfizz/sfizz_wrapper.cpp.o
       > [ 98%] Building CXX object library/src/CMakeFiles/sfizz_import.dir/sfizz/import/foreign_instruments/AudioFile.cpp.o
       > [ 98%] Building CXX object library/src/CMakeFiles/sfizz_import.dir/sfizz/import/foreign_instruments/DecentSampler.cpp.o
       > [100%] Linking CXX static library ../lib/libsfizz.a
       > [100%] Built target sfizz_importy ../lib/libsfizz_import.a
       > [ 14%] Built target sfizz_build
       > [ 14%] Linking CXX static library libjuce.a
       > [ 14%] Built target juce
       > make: *** [Makefile:136: all] Error 2
       For full logs, run:
        nix log /nix/store/gknhhzwaxln6i36mfv921zpr6b164ihr-plugdata-0.9.1.drv
```

### Additional context

_No response_

### System metadata

fetching path input 'path:/nix/store/0yj36irhwn225ywy1saz0gf5wr2ciz50-source'
 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.3, NixOS, 26.05 (Yarara), 26.05.20260105.9f0c42f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.94.0
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux, x86_64-v4-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/emenel/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/emenel/.config/guix/current/etc/xdg/nix/nix.conf:/home/emenel/.guix-home/profile/etc/xdg/nix/nix.conf:/home/emenel/.guix-profile/etc/xdg/nix/nix.conf:/home/emenel/.local/share/flatpak/exports/etc/xdg/nix/nix.conf:/var/lib/flatpak/exports/etc/xdg/nix/nix.conf:/home/emenel/.nix-profile/etc/xdg/nix/nix.conf:/home/emenel/.local/state/nix/profile/etc/xdg/nix/nix.conf:/home/emenel/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/emenel/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/pghc4b78i4wxncxgb4kya393z8hm9ih1-lix-2.94.0/share`
 - nixpkgs: `/nix/store/0yj36irhwn225ywy1saz0gf5wr2ciz50-source`


### Notify maintainers

@PowerUser64 @l1npengtul

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/pl/plugdata/package.nix
+++ b/pkgs/by-name/pl/plugdata/package.nix
@@ -2,47 +2,46 @@
   lib,
   stdenv,
   fetchFromGitHub,
+  fetchpatch2,
   ensureNewerSourcesForZipFilesHook,
   makeDesktopItem,
   copyDesktopItems,
   cmake,
   pkg-config,
   alsa-lib,
+  libX11,
+  libXcursor,
+  libXext,
+  libXinerama,
+  libXrender,
+  libXrandr,
+  libXdmcp,
+  libXtst,
+  xvfb,
   freetype,
-  webkitgtk_4_1,
+  fontconfig,
   zenity,
   curl,
-  xorg,
   python3,
-  libsysprof-capture,
-  pcre2,
-  util-linux,
-  libselinux,
-  libsepol,
-  libthai,
   libxkbcommon,
-  libdatrie,
-  libepoxy,
   libGL,
+  libGLU,
   libjack2,
-  lerc,
-  sqlite,
   expat,
+  webkitgtk_4_1,
   makeWrapper,
+  writableTmpDirAsHomeHook,
   nix-update-script,
 }:
-let
-  version = "0.9.1";
-in
-stdenv.mkDerivation {
+stdenv.mkDerivation (finalAttrs: {
   pname = "plugdata";
-  inherit version;
+  version = "0.9.2";
 
   src = fetchFromGitHub {
     owner = "plugdata-team";
     repo = "plugdata";
-    tag = "v${version}";
-    hash = "sha256-dcggq455lZiwl1lps11fuKX6sx0A8UtFwFoiBJWtwFQ=";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-3ldLM6M54usqjsM9veEctXVa/G14shOdp7Yi9tQi70Y=";
     fetchSubmodules = true;
   };
 
@@ -53,36 +52,29 @@ stdenv.mkDerivation {
     copyDesktopItems
     python3
     makeWrapper
+    writableTmpDirAsHomeHook
   ];
 
   buildInputs = [
     alsa-lib
     curl
     freetype
-    webkitgtk_4_1
-    xorg.libX11
-    xorg.libXcursor
-    xorg.libXext
-    xorg.libXinerama
-    xorg.libXrender
-    xorg.libXrandr
-    xorg.libXdmcp
-    xorg.libXtst
-    xorg.xvfb
-    libsysprof-capture
-    pcre2
-    util-linux
+    fontconfig
     libGL
-    libselinux
-    libsepol
-    libthai
+    libGLU
     libxkbcommon
-    libdatrie
-    libepoxy
-    lerc
+    libX11
+    libXcursor
+    libXext
+    libXinerama
+    libXrender
+    libXrandr
+    libXdmcp
+    libXtst
+    xvfb
     libjack2
     expat
-    sqlite
+    webkitgtk_4_1
   ];
 
   desktopItems = [
@@ -113,9 +105,22 @@ stdenv.mkDerivation {
     ]
   );
 
-  preBuild = ''
-    # fix LV2 build
-    HOME=$(mktemp -d)
+  patches = [
+    # fiddle~.c prevents building with gcc15. Upstream puredata has fixed this issue,
+    # but downstream vendored Plugdata has not implemented the patch yet.
+    # This backports the patch.
+    (fetchpatch2 {
+      url = "https://github.com/pure-data/pure-data/commit/95e4105bc1044cbbcbbbcc369480a77c298d7475.patch";
+      hash = "sha256-siwtizmlAS6fTOE1t+VvDKm1F14YquVlt2QAOKOGX2c=";
+      stripLen = 1;
+      extraPrefix = "Libraries/pure-data/";
+    })
+  ];
+
+  postPatch = ''
+    # Plugdata vendors its own version of Ffmpeg, and this script is used to unpack + build
+    # However, its shebang is broken on nix, so we fix it here.
+    patchShebangs Libraries/pd-else/Source/Shared/ffmpeg/build_ffmpeg.sh
   '';
 
   installPhase = ''
@@ -151,9 +156,9 @@ stdenv.mkDerivation {
     homepage = "https://plugdata.org/";
     license = lib.licenses.gpl3;
     platforms = lib.platforms.linux;
-    maintainers = [
-      lib.maintainers.PowerUser64
-      lib.maintainers.l1npengtul
+    maintainers = with lib.maintainers; [
+      PowerUser64
+      l1npengtul
     ];
   };
-}
+})

